content/blog/vscode-esp-idf-v1.8.0/index.md

```diff
@@ -0,0 +1,137 @@
+---
+title: "ESP-IDF VS Code Extension v1.8.0 release"
+date: 2024-07-12T16:07:09+08:00
+showAuthor: false
+tags: ["News", "ESP-IDF extension", "VSCode", "Visual Studio Code"]
+authors:
+  - "brian-ignacio"
+---
+
+## Introduction
+
+[ESP-IDF VS Code Extension](https://github.com/espressif/vscode-esp-idf-extension) v1.8.0 has been released! It introduces new features and many improvements. If you face any issue or improvement ideas please share them as a [Github issue](https://github.com/espressif/vscode-esp-idf-extension/issues).
+
+This article covers the following key features:
+
+<!-- no toc -->
+- [Eclipse CDT GDB Debug Adapter](#eclipse-cdt-gdb-debug-adapter)
+- [ESP-IDF version switcher](#esp-idf-version-switcher)
+- [Code navigation without full project build](#code-navigation-without-full-project-build)
+- [ESP-IDF Hints viewer](#esp-idf-hints-viewer)
+
+For other updates please review [v1.8.0 Release notes](https://github.com/espressif/vscode-esp-idf-extension/releases/tag/v1.8.0). Now let's look into the key features one by one.
+
+## Eclipse CDT GDB Debug Adapter
+
+First, we will briefly talk about the implementation of debug adapters in VS Code and then will move on to the actual Eclipse CDT GDB Debug Adapter.
+
+### What is a debug adapter in VS Code?
+
+VS Code implements a generic (language-agnostic) _Debugger UI_ that is based on an abstract protocol. That protocol was specifically introduced to communicate with debugger backends. As debuggers typically use their own long-established protocols, some "adapter" is needed to allow debugger UI to communicate with debugger backends. This is described in the [debugging architecture of VS Code](https://code.visualstudio.com/api/extension-guides/debugger-extension#debugging-architecture-of-vs-code) documentation.
+
+This adapter is typically a standalone process with its own protocol. Both the Debug Adapter (DA) and the abstract protocol that is used between the VS Code's Debugger UI and DA -- the [Debug Adapter Protocol](https://microsoft.github.io/debug-adapter-protocol/) (DAP) -- have already been implemented by independent contributors. The diagram below illustrates it better:
+
+{{< figure
+    default=true
+    src="img/debug-arch.webp"
+    alt=""
+    caption=""
```

Comment:
After you added the link to debugger UI docs, I saw that you borrowed the picture from the VS Code documentation. Suggest that we add a link to the source of the image to give credit to the original creator.:

```suggestion
    caption="The image is borrowed from [VS Code documentation](https://code.visualstudio.com/api/extension-guides/debugger-extension#debugging-architecture-of-vs-code)."
```